### ISSUE TITLE
train_interpolate

### ISSUE THREAD

**Issue by marcusbobarUVA** [gifford-lab/prescient#3]
Hi thanks for doing all your great work on this and making the code available. I'm trying to work through your pipeline and I was just wondering if there is there anything different about the data objects used for interpolation? It appears that the names of the interpolation data objects end in "..._impute.pt," but these objects don't seem to be available anywhere for me to inspect them. Can you please advise or possibly provide an example interpolation folder? Thank you for your consideration.

**Comment by sachitsaksena** [comment]
Hi there, thanks for trying out the method! 

To answer your question, **no**, there shouldn't be anything fundamentally different between the data objects used for interpolation Otherwise, a normal data_pt object as described in our documentation will contain keys "x" and "xp" which are the normalized expression and PCA-transformed data, respectively, both split by _timepoint_, which is the same information contained in the "...impute.pt". Feel free to specify what reference to an "..._impute" .pt file you are looking at. 

One thing that is different for interpolation is that the PCA is fit and the model is trained without the held-out timepoint (i.e for a time series with day 2,4,6, the PCA and model use only 2 and 6). You can achieve this by removing the desired held-out time points before running `prescient process_data`. 

P.S. if you are trying to do interpolation, we haven't yet included a dataset agnostic, command-line function that does this end-to-end. We hope to do this in the near future. In the meantime, let me know if you would like some instructions on doing the analysis with the available functions. Alternatively, feel free to take a look at the interpolate functions in [https://github.com/gifford-lab/prescient/blob/master/depr/weinreb.py](url).

**Comment by marcusbobarUVA** [comment]
Hi thanks for the reply! Ok that makes sense. I will make sure to not do the timepoint holdout with the interpolation step. In terms of the impute files, I was referring to the weinreb.py file. Near the top, some directories/files are listed and one of them is ‘IMPUTE_DATA_PATH = "data/Klein2020_impute.pt," which is then found in the ‘train_interpolate’ function.

Regarding instructions on doing the interpolation analysis with the available functions, yes please! From what I can tell after digging through weinreb.py, it seems like I need to first train the model (via args.train) and then subsequently evaluate (via args.evaluate) choosing from evaluate_interpolate_model/data/baseline. Is that correct…or am I way off? Also, it would be great if you could explain when/why I would want to use each of the evaluate model/data/baseline options.

Thank you in advance for your assistance.

-Marcus

**Comment by sachitsaksena** [comment]
You're totally right! For now, you can build a data_pt object without the timepoints you want to hold out (as discussed) and then use `train_model` normally on the subsets of timepoints. Then you can simply use `simulate_trajectories` with time steps up until the held-out timepoint (e.g if you have days 2,4,6 and hold out 4 and you train with a step size of 0.1, you can simulate forward 20 steps).

 You can measure Wasserstein distance between the PCA of the predicted and observed populations 
 using the normalized expression array of the held out time point, your data_pt, and simulate_pt file using a modified version of `evaluate_interpolate_data` that I have now provided in prescient/interpolate/evaluate.py (which is included in [this feature branch](/gifford-lab/prescient/tree/wip-iss3-train-interpolate/prescient/interpolate/evaluate.py) (commit: c273bf7).

To answer your last question, `evaluate_model_baseline` was a wrapper used for evaluating a single baseline model in our paper and  `evaluate_model` is a wrapper for testing multiple different models (with varying architectures and hyperparameters), which also was used for reporting results in the paper. These are features we may include with the aforementioned planned interpolation command line function.
